tensorflow/compiler/xla/service/all_reduce_reassociate.cc

```diff
@@ -141,30 +208,111 @@ StatusOr<bool> AllReduceReassociate::Run(
       if (!ReassociateAllReduceIsProfitable(ar0, ar1, inst)) {
         continue;
       }
-      if (!AreCompatible(ar0, ar1, *kind)) {
+
+      HloInstruction* convert0 = nullptr;
+      HloInstruction* convert1 = nullptr;
+      if (!MatchOperandsToAllReduceWithOptionalConvert(inst, &convert0,
```

Comment:
I think this now needs to work following the recent changes. For example,  between the add and all-reduce, we now allow all sorts of instructions. I'd expect the unit test to now fail with the change.

Reply:
The logic for convert is a separated from other ops now, since converts need to be treated differently. If other ops are matched at the beginning, the match for convert will effectively be no-ops.